### PR TITLE
Created Validator endpoints

### PR DESCRIPTION
**Pull request by DylanGTech**
Continuing from a previous PR (https://github.com/thenewboston-developers/dotnetcore-sdk/pull/32)

This time rebased on the `develop` branch, and the only changes are the addition of the necessary appropriate classes.

### CODE REVIEW

**Comment by DylanGTech** [comment]
That is incredibly strange. It works on my end, and I only changed those files. Seems JSON Deserialization is having a tough time.

**Comment by xmac11** [comment]
Make sure all tests are being run, they are failing locally as well. I'll have a look some time tomorrow, just a few brief comments after glancing at it.
- The appropriate DTOs in `Bank/Models/` and `Validator/Models/` haven't been created. These are going to hold the API responses, not the `ValidatorNode`. Make sure they have the appropriate data types that the API returns. You could name them `BankValidator` and `ValidatorResponseModel`, respectively.
- `PatchValidatorAsync()` hasn't been tested.
- Keep an eye on methods names, for example in your tests you're using `BuildBankTransactionMock()` which is copied from another class. Also always return an interface, when there exists one, not a concrete class.
- Please remove any unnecessary empty lines, for example after `[Fact]` in some tests, or after a method has returned.

**Comment by joshman1019** [comment]
It looks to me like IValidatorService.GetAllValidatorsAsync() should return a Task<IEnumerable<ValidatorNode>> instead of Task<ResponseModel>. When you change that return type and make all necessary adjustments in all other areas, the tests will pass.

**Comment by xmac11** [comment]
These endpoints return paginated results, so `Task<ResponseModel>` is needed.

**Comment by DylanGTech** [comment]
Yeah. I checked the servers to make sure, and they do return paginated results. The types for a couple values were integers instead of strings in the live servers as well, but I left those alone for now to avoid confusion.

**Review by Mirch — APPROVED** [review]

File: src/Thenewboston/Bank/Models/BankValidator.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        public int DefaultTransactionFee { get; set; }
+
+        [JsonProperty(PropertyName = "protocol")]
+        public string Protocol { get; set; }
+
+        [JsonProperty(PropertyName = "ip_address")]
+        public string IpAddress { get; set; }
+
+        [JsonProperty(PropertyName = "port")]
+        public string Port { get; set; }
```
Comment (xmac11) [inline src/Thenewboston/Bank/Models/BankValidator.cs:26]: Let's make this a `nullable int`

File: src/Thenewboston/Bank/Models/BankValidator.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        [JsonProperty(PropertyName = "version")]
+        public string Version { get; set; }
+
+        [JsonProperty(PropertyName = "daily_confirmation_rate")]
+        public string DailyConfirmationRate { get; set; }
```
Comment (xmac11) [inline src/Thenewboston/Bank/Models/BankValidator.cs:32]: @Mirch @joshman1019 what type are we going to be using for coins, `BigInteger?`

File: src/Thenewboston/Validator/Models/ValidatorResponseModel.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        public int DefaultTransactionFee { get; set; }
+
+        [JsonProperty(PropertyName = "protocol")]
+        public string Protocol { get; set; }
+
+        [JsonProperty(PropertyName = "ip_address")]
+        public string IpAddress { get; set; }
+
+        [JsonProperty(PropertyName = "port")]
+        public string Port { get; set; }
```
Comment (xmac11) [inline src/Thenewboston/Validator/Models/ValidatorResponseModel.cs:26]: `nullable int` as well

File: src/Thenewboston/Validator/Models/ValidatorResponseModel.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        [JsonProperty(PropertyName = "version")]
+        public string Version { get; set; }
+
+        [JsonProperty(PropertyName = "daily_confirmation_rate")]
+        public string DailyConfirmationRate { get; set; }
```
Comment (xmac11) [inline src/Thenewboston/Validator/Models/ValidatorResponseModel.cs:32]: Whatever type we decide to use for coins.

File: src/Thenewboston/Validator/Models/ValidatorResponseModel.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    public class ValidatorResponseModel
+    {
+        [JsonProperty(PropertyName = "node_identifier")]
+        public string NodeIdentifier { get; set; }
+
+        [JsonProperty(PropertyName = "account_number")]
+        public string AccountNumber { get; set; }
+
+        [JsonProperty(PropertyName = "default_transaction_fee")]
+        public int DefaultTransactionFee { get; set; }
```
Comment (xmac11) [inline src/Thenewboston/Validator/Models/ValidatorResponseModel.cs:17]: Whatever type we decide to use for coins.

File: src/Thenewboston/Bank/Models/BankValidator.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    public class BankValidator
+    {
+        [JsonProperty(PropertyName = "node_identifier")]
+        public string NodeIdentifier { get; set; }
+
+        [JsonProperty(PropertyName = "account_number")]
+        public string AccountNumber { get; set; }
+
+        [JsonProperty(PropertyName = "default_transaction_fee")]
+        public int DefaultTransactionFee { get; set; }
```
Comment (xmac11) [inline src/Thenewboston/Bank/Models/BankValidator.cs:17]: Whatever type we decide to use for coins.

**Review by xmac11 — COMMENTED** [review]
I'm a bit late, but keeping track of some things that we should change.

**Comment by Mirch** [comment]
> I'm a bit late, but keeping track of some things that we should change.

Sorry, I'm only now seeing this. I'll create an issue for it.

**Pull request merged**